pkg/collection/collector.go

```diff
@@ -0,0 +1,26 @@
+package collection
+
+import (
+	"context"
+	"time"
+
+	"raccoon/pkg/identification"
+	pb "raccoon/pkg/proto"
+)
+
+type CollectRequest struct {
```

Comment:
The main difference is that the `CollecRequest` has all the information required to process a request like an `Identifier` which cannot be taken directly from a gRPC/HTTP request. In future, if we are adding something from application-level it can be put in this struct.

Reply:
In that case we can retire EventsBatch and just use CollectRequest. They both cannot exist.
Alternatively, we can keep EventsBatch as is today as it makes sense as a batch of events and introduce CollectRequest when we see a need for it in future.

Reply:
Done, retired EventsBatch